### ISSUE TITLE
[BUG] Urls with directory tree are not parsed

### ISSUE THREAD

**Issue by velut** [npm/hosted-git-info#57]
# What / Why

For urls that specify a directory tree, for example `https://github.com/pugjs/pug/tree/master/packages/pug`, `hostedGitInfo.fromUrl()` returns `undefined`.

### Current Behavior

```js
const hostedGitInfo = require("hosted-git-info")

const url = "https://github.com/pugjs/pug/tree/master/packages/pug"
const info = hostedGitInfo.fromUrl(url)
console.log(`Info: ${info}`) // Info: undefined
```
Runnable example: https://runkit.com/embed/0c5ftf7jd34d

### Expected Behavior

HostedGitInfo should return the same info provided for shorter urls.

```js
const hostedGitInfo = require("hosted-git-info")

const url = "https://github.com/pugjs/pug/" // Shorter url
const info = hostedGitInfo.fromUrl(url)
console.log(`Info: ${info}`) // Info: git+https://github.com/pugjs/pug.git
```
Runnable example: https://runkit.com/embed/ey9s1kxw3xpv

**Comment by isaacs** [comment]
We're discussing adding syntax for specifying file paths in the git shorthands, but since `https://github.com/pugjs/pug/tree/master/packages/pug` isn't a github repo url, it's not incorrect to return `undefined` for this.

I'd like it to eventually include the path in the shorthand and git committish, so you'd get something like `github:pubjs/pug#master::path:packages/pug` from that longer URL.

In other words, "working as intended", but there's a valid enhancement request here.

**Comment by velut** [comment]
Thank you for the reply.

I reported this issue when I noticed that some packages on npm (e.g., https://registry.npmjs.com/pug and https://registry.npmjs.com/@microsoft/rush) use this URL format in their `repository` field instead of specifying a `directory` sub-field (e.g., https://registry.npmjs.com/react).

**Comment by wraithgar** [comment]
This appears to have been fixed

```javascript
x = hgi.fromUrl("https://github.com/pugjs/pug/tree/master/packages/pug")
GitHost {
  sshtemplate: [Function: sshtemplate],
  sshurltemplate: [Function: sshurltemplate],
  browsetemplate: [Function: browsetemplate],
  browsefiletemplate: [Function: browsefiletemplate],
  docstemplate: [Function: docstemplate],
  httpstemplate: [Function: httpstemplate],
  filetemplate: [Function: filetemplate],
  shortcuttemplate: [Function: shortcuttemplate],
  pathtemplate: [Function: pathtemplate],
  bugstemplate: [Function: bugstemplate],
  hashformat: [Function: formatHashFragment],
  protocols: [ 'git:', 'http:', 'git+ssh:', 'git+https:', 'ssh:', 'https:' ],
  domain: 'github.com',
  treepath: 'tree',
  gittemplate: [Function: gittemplate],
  tarballtemplate: [Function: tarballtemplate],
  extract: [Function: extract],
  type: 'github',
  user: 'pugjs',
  auth: null,
  project: 'pug',
  committish: 'master',
  default: 'https',
  opts: {}
}
x.ssh()
'git@github.com:pugjs/pug.git#master'
```
